webui/react/src/components/kit/Icon.tsx

```diff
@@ -204,21 +204,26 @@ const FourSquaresIcon: React.FC = () => (
 
 export type IconName = (typeof IconNameArray)[number];
 
-export interface Props {
-  color?: 'cancel' | 'error' | 'success';
-  name: IconName;
-  showTooltip?: boolean;
-  size?: IconSize;
-  title: string;
-}
-
-const Icon: React.FC<Props> = ({
-  name,
-  showTooltip = false,
-  size = 'medium',
-  title,
-  color,
-}: Props) => {
+export type Props =
+  | {
+      color?: 'cancel' | 'error' | 'success';
+      name: IconName;
+      showTooltip?: boolean;
+      size?: IconSize;
+      title: string;
+    }
+  | {
```

Comment:
I tried to reduce duplication here using an intersection type but it wouldn't infer correctly at the use site so I just went full-on overload style.